source/client/service_impl.h

```diff
@@ -39,6 +38,12 @@ class ServiceImpl final : public nighthawk::client::NighthawkService::Service,
     logging_context_ = std::make_unique<Envoy::Logger::Context>(
         spdlog::level::from_str("info"), "[%T.%f][%t][%L] %v", log_lock_, false);
   }
+
+  ServiceImpl(std::unique_ptr<Envoy::Logger::Context> logging_context)
```

Comment:
Optional: using a && arg here might save a copy?

Reply:
What's the benefit to choose && comparing to &?

Reply:
Well, my take on it would be:
- With `&param` the smart pointer would be passed by reference, but ownership may or may not be taken by the constructor of `ServiceImpl`. The caller should make no assumptions about that and live with both scenarios after the call. 
- With a `const & std:unique_ptr<...>` parameter it would be clear that no ownership will be taken, but it might be better to unwrap and just pass a reference to the contained object itself as that avoids having to check for nullptr -- or at least delegates that responsibility to the caller.
- With `&&param` it is clear to consuming code that it will no longer own whatever it passes in to such a parameter (irrespective of wether there is a smart pointer involved). And `&&` saves a copy, which isn't a deciding factor in the code line we're talking about here (as it's not frequently called) -- but minding that might be a good habit.

Having said that, it may be good to add a test to cover this constructor, which will then probably also give our friend clang-tidy an opportunity to have a say about it when it inspects how this is going to be used :-) Next to adding actual coverage, which won't hurt either.

Reply:
Thanks so much Otto! This helps a lot!